src/test/java/org/cactoos/list/ImmutableTest.java

```diff
@@ -560,4 +560,30 @@ public void testToString() {
             new IsEqual<>(new ListOf<>("a", "b", "c").toString())
         ).affirm();
     }
+
+    @Test
+    public void subListReturnsListIteratorWithSupportedSet() {
+        new Assertion<>(
+            "subList.listIterator().set() must throw exception",
+            () -> {
+                final ListIterator<String> iterator = new Immutable<>(
+                    new ListOf<String>("one", "two", "three")
+                )
+                    .subList(0, 2)
+                    .listIterator(0);
+                iterator.next();
+                iterator.set("zero");
+                return new Object();
+            },
+            new Throws<>(
+                new MatcherOf<>(
+                    (String msg) -> msg.equals(
+                        "List Iterator is read-only and doesn't allow rewriting items"
```

Comment:
@andreoss `ListIterator` instead of `List Iterator`

Reply:
@fabriciofx It's the exception message